applicationLibrary/molecularDynamics/molecularDynamicsLibrary/LJMultisiteFunctor.h

```diff
@@ -0,0 +1,1520 @@
+
+    if (distanceSquaredCoM > _cutoffSquared) {
+      return;
+    }
+
+    // get number of sites
+    const size_t numSitesA = useMixing ? _PPLibrary->getNumSites(particleA.getTypeId()) : _sitePositionsLJ.size();
+    const size_t numSitesB = useMixing ? _PPLibrary->getNumSites(particleB.getTypeId()) : _sitePositionsLJ.size();
+
+    // get siteIds
+    const std::vector<size_t> siteIdsA =
+        useMixing ? _PPLibrary->getSiteTypes(particleA.getTypeId()) : std::vector<unsigned long>();
+    const std::vector<size_t> siteIdsB =
+        useMixing ? _PPLibrary->getSiteTypes(particleB.getTypeId()) : std::vector<unsigned long>();
+
+    // get unrotated relative site positions
+    const std::vector<std::array<double, 3>> unrotatedSitePositionsA =
+        useMixing ? _PPLibrary->getSitePositions(particleA.getTypeId()) : _sitePositionsLJ;
+    const std::vector<std::array<double, 3>> unrotatedSitePositionsB =
+        useMixing ? _PPLibrary->getSitePositions(particleB.getTypeId()) : _sitePositionsLJ;
+
+    // calculate correctly rotated relative site positions (rotated correctly)
+    const auto rotatedSitePositionsA =
+        autopas::utils::quaternion::rotateVectorOfPositions(particleA.getQ(), unrotatedSitePositionsA);
+    const auto rotatedSitePositionsB =
+        autopas::utils::quaternion::rotateVectorOfPositions(particleB.getQ(), unrotatedSitePositionsB);
+
+    for (int i = 0; i < numSitesA; i++) {
+      for (int j = 0; j < numSitesB; j++) {
+        const auto displacement = autopas::utils::ArrayMath::add(
+            autopas::utils::ArrayMath::sub(displacementCoM, rotatedSitePositionsB[j]), rotatedSitePositionsA[i]);
+        const auto distanceSquared = autopas::utils::ArrayMath::dot(displacement, displacement);
+
+        const auto sigmaSquared =
+            useMixing ? _PPLibrary->getMixingSigmaSquared(siteIdsA[i], siteIdsB[j]) : _sigmaSquared;
+        const auto epsilon24 = useMixing ? _PPLibrary->getMixing24Epsilon(siteIdsA[i], siteIdsB[j]) : _epsilon24;
+        const auto shift6 =
+            applyShift ? (useMixing ? _PPLibrary->getMixingShift6(siteIdsA[i], siteIdsB[j]) : _shift6) : 0;
+
+        // clang-format off
+        // Calculate potential between sites and thus force
+        // Force = 24 * epsilon * (2*(sigma/distance)^12 - (sigma/distance)^6) * (1/distance)^2 * [x_displacement, y_displacement, z_displacement]
+        //         {                         scalarMultiple                                   } * {                     displacement             }
+        // clang-format on
+        const auto invDistSquared = 1. / distanceSquared;
+        const auto lj2 = sigmaSquared * invDistSquared;
+        const auto lj6 = lj2 * lj2 * lj2;
+        const auto lj12 = lj6 * lj6;
+        const auto lj12m6 = lj12 - lj6;  // = LJ potential / (4x epsilon)
+        const auto scalarMultiple = epsilon24 * (lj12 + lj12m6) * invDistSquared;
```

Comment:
Because @FG-TUM  (and other people) recommends it haha.

Other advantages are that it improves sustainability, i.e. if we ever want to change the float type used here. In cases like here, I don't think it would be so bad to replace these `auto`s with `double`s, but in other cases (e.g. with complex types, vectors, arrays, etc) it can make the code cleaner.

If you use a good IDE e.g. Clion it will tell you what the type is, so you don't lose much readability (in my opinion).

This is also why I am pedantic about distance vs displacement it makes it even more clear what is a scalar i.e. `double` and a vector `std::array<double, 3>`.

If you still would like `double` instead of `auto`, I'd suggest talking to @FG-TUM  about this.

Reply:
Alright, my main argument for explicitly using `double` would have been that it makes it easy to spot the scalar quantities.
However, I understand the reason for flexibility and since I'm using CLion, it is fine I guess :D